### PR TITLE
Read only role for application

### LINKED ISSUE #647: New READ_ONLY role
We'd like to introduce a new READ_ONLY role to the AgencyTeamMemberRole enum. Users with this role should be able to _view everything_ in their agency, but should _not be able to make any changes_. That means all text boxes, radio buttons, etc. should be disabled for them. For now, don't worry about adding a tooltip or anything that explains why an input is disabled. If possible, use the native `disabled` property of the input. If there is a custom input doesn't have such a property, don't worry too much about changing it's style or treatment -- just make sure that its handler function exits early without doing anything.

Additionally, if you have a READ_ONLY role, the [badge that shows up in the top-left corner](https://github.com/Recidiviz/justice-counts/blob/c34dbc15bdb8191de32f5576cf688a0db6e3756a/publisher/src/components/Header/Header.tsx#L45) of the header should say Read Only.

The things we need to disable include:

Records Page:
* Can see the Bulk Actions menu, but all of the options should be greyed out
* Can see the New Record button, but it should be disabled
* Can see all Records and click into any one of them

Record Page:
* Can see all existing data
* Cannot add new data or delete or change existing data
* Cannot Save, Publish, or Unpublish a Record (but can see all these buttons, they are just disabled)

Upload Data Page:
* Can view this page
* Cannot drag and drop a file onto the page
* Cannot select a file to upload 

Data Page:
* Can do everything

Metric Configuration Page:
* Can see everything, but cannot make any changes

Settings Page > Your Account:
* Can actually leave this one alone -- it's okay if they make changes here

Settings Page > Agency Settings:
* Can see everything (including Supervision Populations, if applicable)
* Remove all "Edit" links

Settings Page: Team Management:
* Disable the "Invite" button
* Remove the "***" three dot menu that appears next to each name

There may be other inputs that I'm missing -- if so, please go ahead and disable them too! Reach out to @lilidworkin and @mxosman with any questions!

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Disable edits according to task for members with READ_ONLY role

## Type of change
Type: Feature 

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #647 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by lilidworkin — COMMENTED** [review]
@ivan-kishko , this looks great! I'll let @mxosman review the code. I have a couple requests, let me know if any of these are hard:

1. Can we make it possible to still click View/Edit here, and to see the popup, but then to disable the controls on the popup?
Screenshot 2023-05-26 at 10 00 23 AM


2. If it's easy, can we make the controls on the Configuration page look disabled? The buttons and the toggles:
Screenshot 2023-05-26 at 10 00 11 AM

Screenshot 2023-05-26 at 10 00 15 AM

Thank you!

File: publisher/src/components/Settings/AgencySettingsTeamManagement.tsx [hunk 1]
```diff
@@ -245,27 +249,32 @@ export const AgencySettingsTeamManagement = observer(() => {
...
-                    )}
-                  </TeamMemberNameContainer>
-                  <TeamMemberEmailContainer>
-                    {email}
-                    {role !== AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN &&
-                      !userStore.isContributor(agencyId) &&
-                      userStore.email !== email && (
+              ({ name, email, invitation_status, role }) => {
+                const isTeamMemberEditMenuDisabled =
+                  role === AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN ||
```
Comment (mxosman) [inline publisher/src/components/Settings/AgencySettingsTeamManagement.tsx:254]: Double checking this - I think `JUSTICE_COUNTS_ADMIN` role should be able to edit menu, right?

**Review by mxosman — COMMENTED** [review]
Amazing! Looks great, Ivan! There's one more area that we should consider disabling in the `UploadedFiles` component - currently users can still delete files.

In L199:
```
            <Button
              label="Delete"
              onClick={() => deleteUploadedFile(id)}
              labelColor="red"
            />
```

Had just one other question - otherwise code LGTM!

**Comment by mxosman** [comment]
> @ivan-kishko , this looks great! I'll let @mxosman review the code. I have a couple requests, let me know if any of these are hard:
> 
> 1. Can we make it possible to still click View/Edit here, and to see the popup, but then to disable the controls on the popup?
> 2. If it's easy, can we make the controls on the Configuration page look disabled? The buttons and the toggles:


Great suggestions!

File: publisher/src/components/Settings/AgencySettingsTeamManagement.tsx (hunk 1 above)
Comment (ivan-kishko) [reply publisher/src/components/Settings/AgencySettingsTeamManagement.tsx:254]: this is about if user is jc admin you cannot do anything to him even if you are jc admin yourself. Not sure who can confirm that

**Comment by ivan-kishko** [comment]
> Amazing! Looks great, Ivan! There's one more area that we should consider disabling in the `UploadedFiles` component - currently users can still delete files.
> 
> In L199:
> 
> ```
>             <Button
>               label="Delete"
>               onClick={() => deleteUploadedFile(id)}
>               labelColor="red"
>             />
> ```
> 
> Had just one other question - otherwise code LGTM!

[12555c2](https://github.com/Recidiviz/justice-counts/pull/663/commits/12555c2b9edfa754aaf90ac6fa73779692fff5cb)

**Comment by ivan-kishko** [comment]
> @ivan-kishko , this looks great! I'll let @mxosman review the code. I have a couple requests, let me know if any of these are hard:
> 
> 1. Can we make it possible to still click View/Edit here, and to see the popup, but then to disable the controls on the popup?
> 
> Screenshot 2023-05-26 at 10 00 23 AM
> 2. If it's easy, can we make the controls on the Configuration page look disabled? The buttons and the toggles:
> 
> Screenshot 2023-05-26 at 10 00 11 AM Screenshot 2023-05-26 at 10 00 15 AM
> Thank you!

[0127b46](https://github.com/Recidiviz/justice-counts/pull/663/commits/0127b46b2d3edf2c0fa5debfcf45b2ba5715f624)

File: publisher/src/components/Settings/AgencySettingsTeamManagement.tsx (hunk 1 above)
Comment (lilidworkin) [reply publisher/src/components/Settings/AgencySettingsTeamManagement.tsx:268]: I see, I think that's fine for now, thanks for clarifying both!

**Review by lilidworkin — APPROVED** [review]
This looks perfect @ivan-kishko , thanks for making those additional changes! It's ready to go now 🚀

**Pull request merged**